src/include/duckdb/parallel/meta_pipeline.hpp

```diff
@@ -83,6 +86,8 @@ class MetaPipeline : public enable_shared_from_this<MetaPipeline> {
 	PipelineBuildState &state;
 	//! The sink of all pipelines within this MetaPipeline
 	optional_ptr<PhysicalOperator> sink;
+	//! Whether the sink of this pipeline is a join build
+	bool is_join_build;
```

Comment:
nit: can we turn this into an enum, and maybe unify with `recursive_cte`?

Reply:
Thanks for the review! I've turned this into an enum but was not able to unify with `recursive_cte` because we can have join builds within recursive CTEs, and the `MetaPipeline` needs to be annotated with both (i.e., some joins are being performed somewhere inside of the recursion).